gui/src/lianalite/login.rs

```diff
@@ -0,0 +1,564 @@
+        Self::Settings(value)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum Message {
+    View(ViewMessage),
+    OTPRequested(Result<(AuthClient, String), Error>),
+    OTPResent(Result<(), Error>),
+    // wallet_id and result of the connect command.
+    Connected(Result<BackendState, Error>),
+    // redirect to the installer with the remote backend connection.
+    Install(Option<BackendClient>),
+    // redirect to the app runner with the remote backend connection.
+    Run(Result<(BackendWalletClient, api::Wallet), Error>),
+}
+
+#[derive(Debug, Clone)]
+pub enum ViewMessage {
+    RequestOTP,
+    EditEmail,
+    EmailEdited(String),
+    OTPEdited(String),
+    BackToLauncher,
+}
+
+#[derive(Debug, Clone)]
+pub enum BackendState {
+    NoWallet(BackendClient),
+    WalletExists(BackendWalletClient, api::Wallet),
+}
+
+pub struct LianaLiteLogin {
+    pub datadir: PathBuf,
+    pub network: Network,
+
+    wallet_id: String,
+
+    processing: bool,
+    step: ConnectionStep,
+
+    // Error due to connection
+    connection_error: Option<Error>,
+    // Authentification Error
+    auth_error: Option<&'static str>,
+}
+
+pub enum ConnectionStep {
+    CheckingAuthFile,
+    EnterEmail {
+        email: form::Value<String>,
+    },
+    EnterOtp {
+        client: AuthClient,
+        backend_api_url: String,
+        email: String,
+        otp: form::Value<String>,
+    },
+}
+
+impl LianaLiteLogin {
+    pub fn new(datadir: PathBuf, network: Network, settings: Settings) -> (Self, Command<Message>) {
+        match settings
+            .wallets
+            .first()
+            .cloned()
+            .and_then(|w| w.remote_backend_auth)
+            .ok_or(Error::Unexpected(
+                "Missing auth configuration in settings.json".to_string(),
+            )) {
+            Err(e) => (
+                Self {
+                    network,
+                    datadir: datadir.clone(),
+                    step: ConnectionStep::EnterEmail {
+                        email: form::Value::default(),
+                    },
+                    wallet_id: String::new(),
+                    connection_error: Some(e),
+                    auth_error: None,
+                    processing: true,
+                },
+                Command::none(),
+            ),
+            Ok(auth_config) => (
+                Self {
+                    network,
+                    datadir: datadir.clone(),
+                    step: ConnectionStep::CheckingAuthFile,
+                    connection_error: None,
+                    wallet_id: auth_config.wallet_id.clone(),
+                    auth_error: None,
+                    processing: true,
+                },
+                Command::perform(
+                    async move {
+                        let service_config = super::client::get_service_config(network)
+                            .await
+                            .map_err(|e| Error::Unexpected(e.to_string()))?;
```

Comment:
shouldn't we give a clearer error here? i think it's where we 404 atm on mainnet?